test/Microsoft.Azure.WebJobs.Host.UnitTests/Loggers/WebJobsRoleEnvironmentTelemetryInitializerTests.cs

```diff
@@ -2,70 +2,83 @@
+            using (EnvVarHolder.Set(WebJobsRoleEnvironmentTelemetryInitializer.AzureWebsiteSlotName, "Staging"))
+            using (EnvVarHolder.Set(WebJobsRoleEnvironmentTelemetryInitializer.AzureWebsiteCloudRolename, null))
+            {
+                var initializer = new WebJobsRoleEnvironmentTelemetryInitializer();
 
-            var telemetry = new TraceTelemetry();
-            initializer.Initialize(telemetry);
+                var telemetry = new TraceTelemetry();
+                initializer.Initialize(telemetry);
 
-            Assert.Equal("mytestsite-staging", telemetry.Context.Cloud.RoleName);
-            Assert.Equal("mytestsite-staging.azurewebsites.net", telemetry.Context.GetInternalContext().NodeName);
+                Assert.Equal("mytestsite-staging", telemetry.Context.Cloud.RoleName);
+                Assert.Equal("mytestsite-staging.azurewebsites.net", telemetry.Context.GetInternalContext().NodeName);
+            }
         }
 
         [Fact]
         public void Initialize_WithProductionSlot()
         {
-            SetEnvironmentVariables("mytestsite", "Production");
+            using (EnvVarHolder.Set(WebJobsRoleEnvironmentTelemetryInitializer.AzureWebsiteName, "mytestsite"))
+            using (EnvVarHolder.Set(WebJobsRoleEnvironmentTelemetryInitializer.AzureWebsiteSlotName, "Production"))
+            using (EnvVarHolder.Set(WebJobsRoleEnvironmentTelemetryInitializer.AzureWebsiteCloudRolename, null))
+            {
+                var initializer = new WebJobsRoleEnvironmentTelemetryInitializer();
 
-            var initializer = new WebJobsRoleEnvironmentTelemetryInitializer();
+                var telemetry = new TraceTelemetry();
+                initializer.Initialize(telemetry);
 
-            var telemetry = new TraceTelemetry();
-            initializer.Initialize(telemetry);
-
-            Assert.Equal("mytestsite", telemetry.Context.Cloud.RoleName);
-            Assert.Equal("mytestsite.azurewebsites.net", telemetry.Context.GetInternalContext().NodeName);
+                Assert.Equal("mytestsite", telemetry.Context.Cloud.RoleName);
+                Assert.Equal("mytestsite.azurewebsites.net", telemetry.Context.GetInternalContext().NodeName);
+            }
         }
 
-        private static void SetEnvironmentVariables(string websiteName, string slotName)
+        [Fact]
+        public void Initialize_WithWebsiteCloudRoleName()
         {
-            Environment.SetEnvironmentVariable(WebJobsRoleEnvironmentTelemetryInitializer.AzureWebsiteName, websiteName);
-            Environment.SetEnvironmentVariable(WebJobsRoleEnvironmentTelemetryInitializer.AzureWebsiteSlotName, slotName);
-        }
+            var testCloudRoleName = "mycloudrolename";
 
-        public void Dispose()
-        {
-            SetEnvironmentVariables(null, null);
+            using (EnvVarHolder.Set(WebJobsRoleEnvironmentTelemetryInitializer.AzureWebsiteCloudRolename, testCloudRoleName))
```

Comment:
Can you have this test also set `AzureWebsiteName` and `AzureWebsiteSlotName` to something (like the test above). That way we can ensure that `AzureWebsiteCloudRolename` wins if all are specified.

Reply:
Done